stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -947,13 +947,19 @@ dot(x::AbstractVector, transA::Transpose{<:Real}, y::AbstractVector) = adjoint(d
     rank(A::AbstractMatrix; atol::Real=0, rtol::Real=atol>0 ? 0 : n*ϵ)
     rank(A::AbstractMatrix, rtol::Real)
 
-Compute the rank of a matrix by counting how many singular
-values of `A` have magnitude greater than `max(atol, rtol*σ₁)` where `σ₁` is
-`A`'s largest singular value. `atol` and `rtol` are the absolute and relative
+Compute the numerical rank of a matrix by counting how many outputs of 
+`svdvals(A)` are greater than `max(atol, rtol*σ₁)` where `σ₁` is `A`'s largest
+calculated singular value. `atol` and `rtol` are the absolute and relative
 tolerances, respectively. The default relative tolerance is `n*ϵ`, where `n`
 is the size of the smallest dimension of `A`, and `ϵ` is the [`eps`](@ref) of
 the element type of `A`.
 
+!!! note
+`rank` may miscount singular values very close to the threshold due to
+the limits of numerical singular value decomposition. This may result in 
+variations across operating systems and minor versions of Julia. This 
+is especially true for ill conditioned matrices.
```

Comment:
```suggestion
This numerical rank can be a sensitive and imprecise characterization
of the matrix for the case of ill-conditioned matrices with singular
values that are very close to the threshold tolerance `max(atol, rtol*σ₁)`.
Slight perturbations to the tolerances or to the matrix can change
the result of `rank` by pushing one or more singular values across
the threshold, and such variations can even occur due to changes
in floating-point errors between different Julia versions, architectures,
compilers, or operating systems.
```

Reply:
In any case, this needs indentation to be typeset correctly.

Reply:
I omitted "this" in the first sentence to indicate that this is a fundamental problem rather than an implementation problem. I also tweaked the grammar for concision.